src/screens/send/components/fee-options/fee-options.tsx

```diff
@@ -0,0 +1,186 @@
+    blockchainConfig: IBlockchainConfig;
+    hasAdvancedOptions: boolean;
+    selectedPreset: string;
+    showAdvancedOptions: boolean;
+}
+
+interface IReduxProps {
+    chainId: number;
+}
+
+export class FeeOptionsComponent extends React.Component<
+    IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>> & IReduxProps,
+    IState
+> {
+    constructor(
+        props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>> & IReduxProps
+    ) {
+        super(props);
+        const feeOptions = BLOCKCHAIN_INFO[this.props.account.blockchain].feeOptions;
+
+        this.state = {
+            gasPrice: undefined,
+            gasLimit: undefined,
+            blockchainConfig: BLOCKCHAIN_INFO[props.account.blockchain],
+            showAdvancedOptions: false,
+            hasAdvancedOptions: feeOptions.ui.feeComponentAdvanced === 'FeeAdvanced',
+            selectedPreset: feeOptions.ui.defaultPreset
+        };
+        this.estimatedFees();
+    }
+
+    public estimatedFees = async () => {
+        const blockchainInstance = getBlockchain(this.props.account.blockchain);
+        const fees = await blockchainInstance
+            .getClient(this.props.chainId)
+            .estimateFees(this.props.account.address, this.props.toAddress);
+
+        this.setState({
+            gasPrice: fees.gasPrice,
+            gasLimit: fees.gasLimit
+        });
+        this.props.calculatedFees(fees.gasPrice, fees.gasLimit);
+    };
+
+    public onSelectFeePreset = (key: string) => {
+        this.setState({
+            selectedPreset: key,
+            gasPrice: this.state.blockchainConfig.feeOptions.defaults.gasPricePresets[key],
+            gasLimit: this.state.blockchainConfig.feeOptions.defaults.gasLimit
+        });
+        this.props.calculatedFees(
+            this.state.blockchainConfig.feeOptions.defaults.gasPricePresets[key],
+            this.state.blockchainConfig.feeOptions.defaults.gasLimit
+        );
+    };
+
+    public onInputAdvancedFees = (gasPrice: BigNumber, gasLimit: BigNumber) => {
+        this.setState({
+            gasPrice,
+            gasLimit
+        });
+        this.props.calculatedFees(gasPrice, gasLimit);
+    };
+
+    public onAdvancedButton = () => {
+        const currentState = this.state.showAdvancedOptions;
+        this.setState({
+            showAdvancedOptions: !currentState
+        });
+    };
+
+    public renderSimpleFees() {
+        const styles = this.props.styles;
+
+        if (this.state.blockchainConfig.feeOptions.ui.feeComponent === 'FeeTotal') {
+            return (
+                <FeeTotal amount={this.state.gasPrice} blockchain={this.props.account.blockchain} />
+            );
+        } else if (this.state.blockchainConfig.feeOptions.ui.feeComponent === 'FeePresets') {
+            const presets = this.state.blockchainConfig.feeOptions.defaults.gasPricePresets;
+            return (
+                <View style={styles.containerPresets}>
```

Comment:
do we need the `View` ?

Reply:
In case the number of presets ar odd then the last one will display on full width